nav2_simple_commander/nav2_simple_commander/robot_navigator.py

```diff
@@ -0,0 +1,408 @@
+            self.info('Clear local costmaps service not available, waiting...')
+        req = ClearEntireCostmap.Request()
+        future = self.clear_costmap_local_srv.call_async(req)
+        rclpy.spin_until_future_complete(self, future)
+        return
+
+    def clearGlobalCostmap(self):
+        while not self.clear_costmap_global_srv.wait_for_service(timeout_sec=1.0):
+            self.info('Clear global costmaps service not available, waiting...')
+        req = ClearEntireCostmap.Request()
+        future = self.clear_costmap_global_srv.call_async(req)
+        rclpy.spin_until_future_complete(self, future)
+        return
+
+    def getGlobalCostmap(self):
+        while not self.get_costmap_global_srv.wait_for_service(timeout_sec=1.0):
+            self.info('Get global costmaps service not available, waiting...')
+        req = GetCostmap.Request()
+        future = self.get_costmap_global_srv.call_async(req)
+        rclpy.spin_until_future_complete(self, future)
+        return future.result().map
+
+    def getLocalCostmap(self):
+        while not self.get_costmap_local_srv.wait_for_service(timeout_sec=1.0):
+            self.info('Get local costmaps service not available, waiting...')
+        req = GetCostmap.Request()
+        future = self.get_costmap_local_srv.call_async(req)
+        rclpy.spin_until_future_complete(self, future)
+        return future.result().map
+
+    def lifecycleStartup(self):
+        self.info('Starting up lifecycle nodes based on lifecycle_manager.')
+        srvs = self.get_service_names_and_types()
+        for srv in srvs:
+            if srv[1][0] == 'nav2_msgs/srv/ManageLifecycleNodes':
+                srv_name = srv[0]
+                self.info('Starting up ' + srv_name)
+                mgr_client = self.create_client(ManageLifecycleNodes, srv_name)
+                while not mgr_client.wait_for_service(timeout_sec=1.0):
+                    self.info(srv_name + ' service not available, waiting...')
+                req = ManageLifecycleNodes.Request()
+                req.command = ManageLifecycleNodes.Request().STARTUP
+                future = mgr_client.call_async(req)
+
+                # starting up requires a full map->odom->base_link TF tree
+                # so if we're not successful, try forwarding the initial pose
+                while True:
+                    rclpy.spin_until_future_complete(self, future, timeout_sec=0.10)
+                    if not future:
+                        self._waitForInitialPose()
+                    else:
+                        break
+        self.info('Nav2 is ready for use!')
+        return
+
+    def lifecycleShutdown(self):
+        self.info('Shutting down lifecycle nodes based on lifecycle_manager.')
+        srvs = self.get_service_names_and_types()
+        for srv in srvs:
+            if srv[1][0] == 'nav2_msgs/srv/ManageLifecycleNodes':
+                srv_name = srv[0]
+                self.info('Shutting down ' + srv_name)
```

Comment:
So @klintan suggestion would look like this right?

```suggestion
        for srv_name, srv_type in self.get_service_names_and_types():
            if srv_type[0] == 'nav2_msgs/srv/ManageLifecycleNodes':
                self.info('Shutting down ' + srv_name)
```

Reply:
Will do!